### PR TITLE
Jbaker/date range support

### PR DESCRIPTION
**Pull request by jason-capsule42**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #106, #112

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: package.json
Comment (blackfalcon) [inline package.json:25]: I am confused... I thought that this was updated to a Lit 2.0 element? The Lit dependency is being supported by the wc-range-datepicker dependency.

File: src/auro-calendar-month.js
Comment (blackfalcon) [inline src/auro-calendar-month.js:4]: You can't import from outside the scope of the element.

File: src/auro-calendar.js
Comment (blackfalcon) [inline src/auro-calendar.js:5]: You can't import from outside the scope of the element.

File: src/auro-datepicker.js
Comment (blackfalcon) [inline src/auro-datepicker.js:9]: This is importing the legacy Lit Element?

**Review by blackfalcon — DISMISSED** [review]
I am seeing that the bundle version of the element is not working. There are references to direct JS files that are outside the scope of the element, this may be the cause of that as well as the issues with generating the API documentation. 

WCA Analize cannot import outside the scope of the element.

File: src/auro-datepicker.js
Comment (jason-capsule42) [reply src/auro-datepicker.js:9]: Good catch, this is updated now.

File: package.json
Comment (jason-capsule42) [reply package.json]: Good catch, this is updated now.

File: src/auro-datepicker.js
Comment (jason-capsule42) [reply src/auro-datepicker.js]: Good catch, this is updated now.

**Comment by blackfalcon** [comment]
The commits early in the history using `BREAKING CHANGE` as the type will not create a MAJOR release. Please see the rules here https://www.conventionalcommits.org/en/v1.0.0/#specification

According to conventional commits, you can use a `!` to designate a breaking change, but this example I am not finding in the semantic-release docs. https://semantic-release.gitbook.io/semantic-release/

The following example is known to work. 

```
feat: move feature from Node 6 to Node 8 support

BREAKING CHANGE: This feature requires a major update to Node support due to deprecated Node features
```

File: README.md
Comment (blackfalcon) [inline README.md:97]:
```suggestion
<script src="https://cdn.jsdelivr.net/npm/@alaskaairux/auro-icon@latest/dist/auro-icon__bundled.js" type="module"></script>

```

File: apiExamples/error.js
Comment (blackfalcon) [inline apiExamples/error.js:5]: remove commented code

File: apiExamples/error.js
Comment (blackfalcon) [inline apiExamples/error.js:11]: remove commented code

File: demo/api.js
Comment (blackfalcon) [inline demo/api.js:7]: :+1:

File: demo/index.html
Comment (blackfalcon) [inline demo/index.html:33]:
```suggestion
      import 'https://cdn.jsdelivr.net/npm/marked@4.3.0/marked.min.js';
```

Consider updating all CDN references.

File: package.json
Comment (blackfalcon) [inline package.json:72]:
```suggestion

```

See for how to override nested dependency https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides

File: src/auro-calendar-cell.js
Comment (blackfalcon) [inline src/auro-calendar-cell.js:80]: This and all code is missing JSDoc comments?

File: src/auro-calendar-month.js
Comment (blackfalcon) [inline src/auro-calendar-month.js:60]: Same JSDoc comment.

File: src/auro-calendar.js
Comment (blackfalcon) [inline src/auro-calendar.js:88]: This and `renderCalendar` are the only code blocks that are properly documented.

File: src/style-auro-calendar-cell.scss
Comment (blackfalcon) [inline src/style-auro-calendar-cell.scss:31]: Really wish we had a consistent token for this.

File: src/style-auro-calendar-cell.scss
Comment (blackfalcon) [inline src/style-auro-calendar-cell.scss:43]:
```suggestion
      width: var(--auro-size-600);
      height: var(--auro-size-600);
```

File: src/style-auro-calendar-cell.scss
Comment (blackfalcon) [inline src/style-auro-calendar-cell.scss:51]: ???

File: src/style-auro-calendar-month.scss
Comment (blackfalcon) [inline src/style-auro-calendar-month.scss:41]: That's an odd number? Can we make this `var(--auro-size-600)`?

File: src/style-auro-calendar.scss
Comment (blackfalcon) [inline src/style-auro-calendar.scss:12]:
```suggestion
:root {
  --mobile-header-height: 86px;
  --mobile-footer-height: 200px;
}
```

Preferred that we move to CSS variables.

File: src/style-auro-calendar.scss
Comment (blackfalcon) [inline src/style-auro-calendar.scss:109]:
```suggestion
      height: var(--auro-size-300);
```

File: src/style-auro-calendar.scss
Comment (blackfalcon) [inline src/style-auro-calendar.scss:128]:
```suggestion
  background: linear-gradient(180deg, var(--auro-color-base-white) 0%, var(--auro-color-background-lightest) 51.56%);
```

File: src/style-auro-calendar.scss
Comment (blackfalcon) [inline src/style-auro-calendar.scss:138]: We should make this a variable at the top of the document. Easier to address if needed to update and reuse if needed.

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: package.json
Comment (jason-capsule42) [reply package.json:76]: I'm not sure why this needs to change? I just did a version update across deps using `npm-check` and this got looped in.

File: src/auro-calendar-cell.js
Comment (jason-capsule42) [reply src/auro-calendar-cell.js:80]: The functions here are just updating what was already in `wc-range-datepicker`. It's mostly a copy of their code with some slight tweaks to get the templates working the way we want.

I propose we clean up all the JSDoc for their functions when we cleanup all the actual JS code in a future release.

File: src/auro-calendar-month.js
Comment (jason-capsule42) [reply src/auro-calendar-month.js:60]: Same comment as above about copying code from `wc-range-datepicker`.

File: src/style-auro-calendar-cell.scss
Comment (jason-capsule42) [reply src/style-auro-calendar-cell.scss:51]: This is per discussion with @jason-capsule42. The token for hover has transparency. However, that causes issues with the pseudo element box on dates that are `inRange` when selecting a from and to date. We had to switch to a solid color to account for this. I left the comment on there are a sort of callout but I'll add more language to it.

File: src/style-auro-calendar-month.scss
Comment (jason-capsule42) [reply src/style-auro-calendar-month.scss:41]: It's odd but it's what makes the layout work for design from Kayla. Any opinion to add here, @leeejune?

File: src/style-auro-calendar.scss
Comment (jason-capsule42) [reply src/style-auro-calendar.scss:12]: Declaring the variable this way at the style.scss level does not seem to actually create the variable for use within the file.

Also, using a variable name here seems questionable to me as we would need to rename all these to something we can be more sure is unique. Any number of components might end up declaring `--mobile-footer-height`.

Comment (jason-capsule42) [reply src/style-auro-calendar.scss:12]: I suggest issues against the design be opened to get these sections of the layout built using the design tokens. It's the kind of thing June is good at making sure she does on hers but this one was done outside our team.

We can do a UI iteration later to refine the layout a bit.

**Review by blackfalcon — CHANGES_REQUESTED** [review]
See the new issues I created for additional work that should be addressed within this milestone. 

https://github.com/AlaskaAirlines/auro-datepicker/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22Date+range+selection+feature+%22+label%3Anot-reviewed

**Comment by jordanjones243** [comment]
> I am seeing an issue in the log where the bundled CDN version is not being produced correctly.
> 
> ```
> ./src/auro-datepicker.js → dist/...
> (!) Unresolved dependencies
> https://rollupjs.org/guide/en/#warning-treating-module-as-external-dependency
> lit-html/is-server.js (imported by node_modules/lit/index.js)
> (!) Unused external imports
> LitElement,html,css imported from external module "lit-html/is-server.js" but never used in "node_modules/lit/index.js"
> created dist/ in 6.1s
> ```
> 
> Has the CDN version been tested and verified it's working?
> 
> I'd consider the updates we made to the Generator https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/rollup.config.mjs

What are you doing to run into this issue? I am unable to find/reproduce this error.

File: package.json
Comment (blackfalcon) [inline package.json:24]: Revert back to v2.13.1. A bug was introduced with the last release that is causing an accessibility violation

https://github.com/AlaskaAirlines/auro-datepicker/actions/runs/4962714746/jobs/8881107808#step:5:121

```
❌ auro-datepicker > auro-datepicker is accessible
      Error: Accessibility Violations
      ---
      Rule: label-title-only
      Impact: serious
      Form elements should have a visible label (https://dequeuniversity.com/rules/axe/4.7/label-title-only?application=axeAPI)
      
      Issue target: auro-datepicker,auro-input,input
      Context: <input part="input" class=" inputIcon inputElement " id="" type="month-day-year" inputmode="numeric" aria-describedby="0ecv6d" aria-invalid="true" placeholder="MM/DD/YYYY" lang="">
      Fix all of the following:
        Only title used to generate label for form element
      ---
        at processResults (node_modules/chai-a11y-axe/src/accessible.js:87:15)
        at node_modules/chai-a11y-axe/src/accessible.js:[123](https://github.com/AlaskaAirlines/auro-datepicker/actions/runs/4962714746/jobs/8881107808#step:5:124):7
        at async o.<anonymous> (test/auro-datepicker.test.js:10:5)
```

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: package.json
Comment (blackfalcon) [reply package.json:24]: What is weird is that this issues only comes up when you move to the latest version of auro-input. Nothing functional changed with this version. 

If you could shed any light on why this may be happening inside auro-datepicker it would be appreciated.

**Comment by jason-capsule42** [comment]
> I am seeing an issue in the log where the bundled CDN version is not being produced correctly.
> 
> ```
> ./src/auro-datepicker.js → dist/...
> (!) Unresolved dependencies
> https://rollupjs.org/guide/en/#warning-treating-module-as-external-dependency
> lit-html/is-server.js (imported by node_modules/lit/index.js)
> (!) Unused external imports
> LitElement,html,css imported from external module "lit-html/is-server.js" but never used in "node_modules/lit/index.js"
> created dist/ in 6.1s
> ```
> 
> Has the CDN version been tested and verified it's working?
> 
> I'd consider the updates we made to the Generator https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/rollup.config.mjs

@blackfalcon I have updated the rollup config to match the latest from our generator. I rebuilt, bundled and tested the localhost with that bundle and it all appears to work correctly for me now.

File: package.json
Comment (blackfalcon) [reply package.json:67]: @jason-capsule42 this is what we did for the Generator https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/package.json#L74-L78

From what I understand, this ensures that a nested dependency of another dependency, e.g. wc-sass-render, will run the version of the dependency we want. 

We have no control over what version of Sass wc-sass-render is used and the maintainer never updates it.

File: src/auro-calendar-cell.js
Comment (blackfalcon) [reply src/auro-calendar-cell.js:80]: @jason-capsule42 is there an issue created to address the work for adding comments to the functions?

**Review by blackfalcon — APPROVED** [review]
Approving with cavitate that remaining open questions will be resolved or new issues created to address post-release.

**Comment by leeejune** [comment]
@blackfalcon The token value can be replaced to elevation-200

**Pull request merged**
